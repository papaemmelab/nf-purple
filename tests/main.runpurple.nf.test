nextflow_process {

    name "Test Process runPurple"
    script "main.nf"
    process "runPurple"

    test("Should run Purple with failures") {
        setup {
            run("runAmber") {
                script "main.nf"
                process {
                    """
                    input[0] = Channel.value(params.tumor)
                    input[1] = Channel.value(params.normal)
                    input[2] = Channel.fromPath(params.tumorBam)
                    input[3] = Channel.fromPath(params.normalBam)
                    """
                }
            }
            run("runCobalt") {
                script "main.nf"
                process {
                    """
                    input[0] = Channel.value(params.tumor)
                    input[1] = Channel.value(params.normal)
                    input[2] = Channel.fromPath(params.tumorBam)
                    input[3] = Channel.fromPath(params.normalBam)
                    """
                }
            }
        }
        when {
            params {
                tumor = "TEST_TUMOR"
                normal = "TEST_NORMAL"
                tumorBam = "${projectDir}/tests/data/tumor.bam"
                normalBam = "${projectDir}/tests/data/normal.bam"
                outdir = "${projectDir}/tests/outdir"
                refGenome = "${projectDir}/tests/data/reference.fasta"
            }
            process {
                """
                input[0] = Channel.value(params.tumor)
                input[1] = Channel.value(params.normal)
                input[2] = runAmber.out
                input[3] = runCobalt.out
                input[4] = "${params.outdir}/cobalt"
                """
            }
        }
        then {
            // Purple process should fail when trying to create the segmentation plots
            assert process.failed
            assert process.exitStatus == 1
            assert process.errorReport.contains("[ERROR]")
        }

    }

}
