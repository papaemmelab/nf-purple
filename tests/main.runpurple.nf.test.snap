{
    "Should run Purple with failures": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "10": [
                    
                ],
                "11": [
                    
                ],
                "12": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    
                ],
                "purple_circos_png": [
                    
                ],
                "purple_cnv_gene_tsv": [
                    
                ],
                "purple_cnv_somatic_tsv": [
                    
                ],
                "purple_copynumber_png": [
                    
                ],
                "purple_input_png": [
                    
                ],
                "purple_map_png": [
                    
                ],
                "purple_purity_range_png": [
                    
                ],
                "purple_purity_range_tsv": [
                    
                ],
                "purple_purity_tsv": [
                    
                ],
                "purple_qc": [
                    
                ],
                "purple_segment_png": [
                    
                ],
                "purple_segment_tsv": [
                    
                ],
                "purple_somatic_clonality_tsv": [
                    
                ]
            }
        ],
        "timestamp": "2024-01-23T11:18:57.811454"
    }
}